quix-frontend/client/src/lib/sql-autocomplete/adapter/sql-autocpmplete-adapter-utills.ts

```diff
@@ -0,0 +1,304 @@
+    return "map";
+  }
+
+  if (trinoObjectAsString.substring(index, index + 6) === StartingOptions.array) {
+    index = findClosingParentheses(trinoObjectAsString, index);
+    return "array";
+  }
+    return trinoObjectAsString;
+}
+
+export function processRow(trinoObjectAsString: string, index: number): object {
+    const finalObject: object = {};
+    let key = "";
+    let value = "";
+    let objectToInsert;
+    let firstword = true;
+  
+    while (index <= trinoObjectAsString.length) {
+      switch (trinoObjectAsString.charAt(index)) {
+        case SpecialCharacters.openParenthesis: {
+          const start = findBeginningOfWord(trinoObjectAsString, index);
+          const result = handleStartingOptions(trinoObjectAsString, start, index);
+          objectToInsert = result.objectToInsert;
+          finalObject[key] = result.type;
+          index = result.newIndex;
+          break;
+        }
+        case SpecialCharacters.closedParenthesis: {
+          checkKey(key, index);
+          finalObject[key] = objectToInsert || value;
+          return finalObject;
+        }
+        case SpecialCharacters.comma: {
+          if (value === "") {
+            throw new Error(`Error at index: ${trinoObjectAsString.substring(index-5,index)}, type expected before comma`);
+          }
+          finalObject[key] = objectToInsert || value;
+          key = "";
+          value = "";
+          objectToInsert = undefined;
+          firstword = true;
+          break;
+        }
+        case SpecialCharacters.space: {
+          firstword = isComma(trinoObjectAsString, index);
+          break;
+        }
+        default: {
+          const charIndex = trinoObjectAsString.charAt(index);
+          if (firstword) {
+            key += charIndex;
+            checkKey(key, index);
+          } else {
+            value += charIndex;
+          }
+        }
+      }
+      index++;
+    }
+    return {};
+  }
+
+export function findBeginningOfWord(str: string, index: number): number {
+  while (index > 0 && str[index] !== ' ') {
+    index--;
+  }
+  return index;
+}
+
+function isComma(trinoObjectAsString: string, counter: number): boolean {
+  return trinoObjectAsString.charAt(counter - 1) === SpecialCharacters.comma;
+}
+
+function startsWithRow(trinoObjectAsString: string, counter: number) {
+  return trinoObjectAsString.substring(counter, counter + 4) === StartingOptions.row;
+}
+
+function getOperator(trinoObjectAsString: string, start: number , end: number) {
+  return trinoObjectAsString.substring(start+1, end+1); //we are in middle of string start is ' ' 
+}
+
+function findClosingParentheses(trinoObjectAsString: string, counter: number): number { //rename
+  let parenthesis = 0;
+  for (let i = counter; i < trinoObjectAsString.length; i++) {
+    if (trinoObjectAsString.charAt(i) === '(') {
+      parenthesis++;
+    }
+    if (trinoObjectAsString.charAt(i) === ')') {
+      parenthesis--;
+    }
```

Comment:
You have enum for that.. `'(',  ')'`
Use it!

Reply:
fixed,
and changed is comma also.
say no to racism